TypeCobol/Compiler/Sql/Scanner/SqlScanner.cs

```diff
@@ -50,46 +80,242 @@ public override Token GetTokenStartingFrom(int startIndex)
+                //Check IntegerLiteral range
+                if (literalValueLength > 31)
+                {
+                    //Out of range
+                    this.tokensLine.AddDiagnostic(MessageCode.SyntaxErrorInParser, token,
+                        "Decimal is too big : " + literalValue.Number);
+                }
+            }
+            void CheckFloatingPointLiteral()
+            {
+                Debug.Assert(token.LiteralValue is FloatingPointLiteralTokenValue);
+                var literalValue = (FloatingPointLiteralTokenValue)token.LiteralValue;
+
+                //Check exponent length
+                var exponent = literalValue.Exponent.Number;
+                int exponentLength =
+                    exponent.ToString().Length -
+                    (exponent.Sign == -1 ? 1 : 0); //Check digits only, so remove '-' at beginning of negative numbers
+                var mantissa = literalValue.Mantissa.Number;
+                int mantissaLength =
+                    mantissa.ToString().Length;
+                if (exponentLength > 2 || mantissaLength > 17)
+                {
+                    token = new Token(TokenType.SQL_DecimalFloatingPointLiteral, startIndex, currentIndex - 1,
+                        tokensLine)
+                    {
+                        LiteralValue =
+                            new Sql.Scanner.DecimalFloatingPointLiteralTokenValue(literalValue)
+                    };
+                    CheckDecimalFloatingPointLiteral(exponentLength, mantissaLength);
+                }
+            }
+
+            void CheckDecimalFloatingPointLiteral(int exponentLength, int mantissaLength)
+            {
+                Debug.Assert(token.LiteralValue is DecimalFloatingPointLiteralTokenValue);
+                if (exponentLength > 4 || mantissaLength > 34)
+                {
+                    tokensLine.AddDiagnostic(MessageCode.InvalidExponentInFloatingPointLiteral, token);
+                }
+            }
+        }
+
+        private bool TryScanDecimalFloatingPointSpecialValue(int startIndex, bool hasExplicitSign,
+            out DecimalFloatingPointLiteralTokenValue decimalFloatingPointLiteralTokenValue)
+
+        {
+            decimalFloatingPointLiteralTokenValue = null;
+            for (int index = startIndex; index <= lastIndex && char.IsLetter(line[index]); index++)
+            {
+            }
+
+            string text = null;
+            text = hasExplicitSign ? line.Substring(startIndex + 1, currentIndex - startIndex - 1) : line.Substring(startIndex, currentIndex - startIndex);
+            var tryToGetValue = _SpecialValues.TryGetValue(text, out var outPut);
+            if (tryToGetValue != true) return false;
+            //negative
+            decimalFloatingPointLiteralTokenValue = new DecimalFloatingPointLiteralTokenValue(new DecimalFloatingPointSpecialValue(hasExplicitSign, outPut));
+            return true;
```

Comment:
```csharp
if (_SpecialValues.TryGetValue(text, out var specialValueType)
{
    decimalFloatingPointLiteralTokenValue = new DecimalFloatingPointLiteralTokenValue(new DecimalFloatingPointSpecialValue(hasExplicitSign, specialValueType));
            return true;
}

decimalFloatingPointLiteralTokenValue = null;
return false;
```

and you can remove the `decimalFloatingPointLiteralTokenValue = null;` at beginning.